### ISSUE TITLE
[REST] : Missing endpoint for retrieving folder tree 

### ISSUE THREAD

**Issue by luisbacca** [dotCMS/core#18964]
Reported via Support: [101712](https://dotcms.zendesk.com/agent/tickets/101712)
Reproducible in dotCMS 5.2.8, and latest master (5.3.1)

### Describe the bug
Dotcms doesn't have a way to get the folder tree for any given folder despite the value of 'Show on Menu' attribute. The navtool viewtool requires that a folder is marked as 'Show on Menu' or it won't be returned from this call. The closer method we have to this is `/api/v1/folder/sitename/{siteName}/uri/{uri}` exposed on FolderResource, however this method doesn't return the right tree for the specify uri. 

### Expected behavior
Expose an endpoint that returns a folder structure for any given folder. This feature will only return the requested folder and all folder children recursively. It will not show any information related to the files inside those folder.

### Sample Json Response
```json
{
    "entity": {
        "path": "/parent/self/",
        "children": [
            {
                "path": "/parent/self/child1/",
                "children": [
                    {
                        "path": "/parent/self/child1/grandchild1/",
                        "children": []
                    },
                    {
                        "path": "/parent/self/child1/grandchild2/",
                        "children": []
                    },
                    {
                        "path": "/parent/self/child1/grandchild3/",
                        "children": []
                    }
                ]
            },
            {
                "path": "/parent/self/child2/",
                "children": [
                    {
                        "path": "/parent/self/child2/grandchild1/",
                        "children": []
                    },
                    {
                        "path": "/parent/self/child2/grandchild2/",
                        "children": []
                    },
                    {
                        "path": "/parent/self/child2/grandchild3/",
                        "children": []
                    }
                ]
            }
        ]
    },
    "errors": [],
    "i18nMessagesMap": {},
    "messages": [],
    "permissions": []
}
```

## Notes 
The below json is just a mock up. The final json will have more datails about the folder like identifier, inode, title, etc.
My proposal url will look like ```api/v2/folder/sitename/{sitename}/uri/{uri}``` but this is open for discussion/changes. 

## Acceptance Criteria

1. Unit test

**Comment by luisbacca** [comment]
PR : https://github.com/dotCMS/core/pull/18976

**Comment by erickgonzalez** [comment]
**Note to Doc and QA:**

New Endpoint to get Folder Info and SubFolders: `/api/v1/folder/hostId/{hostId}/path/{path}`

It respects permissions, so if the user does not have permissions over a folder or subfolder will not be shown.

E.g 

`/api/v1/folder/hostId/48190c8c-42c4-46af-8d1a-0cd5db894797/path/images`

The response will look like:
```
{
    "entity": {
        "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
        "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.jpeg, *.svp, *.webp, *.ico",
        "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
        "iDate": 1540474702297,
        "identifier": "8ddef32b-89f3-47ab-80bb-a46ec89046db",
        "inode": "2ad0dd36-5b07-41ac-b9f5-c7c54085ac58",
        "modDate": 1571758192527,
        "name": "images",
        "path": "/images/",
        "showOnMenu": false,
        "sortOrder": 0,
        "subFolders": [
            {
                "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                "iDate": 1568916271024,
                "identifier": "2fa0a962-d913-41a2-969d-d115094807f0",
                "inode": "9716b18a-8b2f-44de-8c10-13f886c6f119",
                "modDate": 1568916673160,
                "name": "blogs",
                "path": "/images/blogs/",
                "showOnMenu": false,
                "sortOrder": 0,
                "subFolders": [],
                "title": "blogs",
                "type": "folder"
            },
            {
                "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                "iDate": 1562786070981,
                "identifier": "6d2a7c3a-1e4f-4de5-9ba1-d4ca973b8ec2",
                "inode": "87314c30-cc80-4eb8-a7d6-dbabbb7afc45",
                "modDate": 1568916698031,
                "name": "events",
                "path": "/images/events/",
                "showOnMenu": false,
                "sortOrder": 0,
                "subFolders": [],
                "title": "events",
                "type": "folder"
            },
            {
                "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                "iDate": 1562870298935,
                "identifier": "8de2cb2b-2072-46ec-88c0-93206b2e4dec",
                "inode": "c0603b7a-b34a-466d-9dcc-590b35d07cbc",
                "modDate": 1568916704649,
                "name": "gallery",
                "path": "/images/gallery/",
                "showOnMenu": false,
                "sortOrder": 0,
                "subFolders": [
                    {
                        "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                        "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                        "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                        "iDate": 1567001241362,
                        "identifier": "b5713c0a-4822-487c-b91b-ca1977e4d24b",
                        "inode": "c5eaf1a7-03c0-4f04-9e55-5f6c45d14e47",
                        "modDate": 1568916711397,
                        "name": "beach",
                        "path": "/images/gallery/beach/",
                        "showOnMenu": false,
                        "sortOrder": 0,
                        "subFolders": [],
                        "title": "beach",
                        "type": "folder"
                    },
                    {
                        "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                        "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                        "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                        "iDate": 1567001271012,
                        "identifier": "740444be-cef8-4d87-a710-b07303cb3d4c",
                        "inode": "9283ed27-c5fb-43d8-ba38-e7b2b628efcf",
                        "modDate": 1568916717664,
                        "name": "surfing",
                        "path": "/images/gallery/surfing/",
                        "showOnMenu": false,
                        "sortOrder": 0,
                        "subFolders": [],
                        "title": "surfing",
                        "type": "folder"
                    },
                    {
                        "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                        "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                        "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                        "iDate": 1567001201806,
                        "identifier": "18e0b4f0-7eb9-4d76-972d-0bee6a080c83",
                        "inode": "3e924cd2-3bae-4e34-afbd-af5bb489992b",
                        "modDate": 1570030383491,
                        "name": "winter",
                        "path": "/images/gallery/winter/",
                        "showOnMenu": false,
                        "sortOrder": 0,
                        "subFolders": [],
                        "title": "winter",
                        "type": "folder"
                    }
                ],
                "title": "gallery",
                "type": "folder"
            },
            {
                "defaultFileType": "d5ea385d-32ee-4f35-8172-d37f58d9cd7a",
                "filesMasks": "*.jpg, *.gif, *.bmp, *.png, *.svg, *.webp, *.jpeg",
                "hostId": "48190c8c-42c4-46af-8d1a-0cd5db894797",
                "iDate": 1564156741722,
                "identifier": "0668187e-aae0-4795-898d-a913b0eaec15",
                "inode": "d708807f-6dbc-4d84-add4-6caec90de177",
                "modDate": 1568916731208,
                "name": "pages",
                "path": "/images/pages/",
                "showOnMenu": false,
                "sortOrder": 0,
                "subFolders": [],
                "title": "pages",
                "type": "folder"
            }
        ],
        "title": "images",
        "type": "folder"
    },
    "errors": [],
    "i18nMessagesMap": {},
    "messages": [],
    "permissions": []
}
```

**Comment by bryanboza** [comment]
Fixed, tested on release-5.3.8 // Postgres // FF
